packages/flutter_tools/gradle/aar_init_script.gradle

```diff
@@ -8,7 +8,24 @@ import org.gradle.api.Project
 import org.gradle.api.artifacts.Configuration
 import org.gradle.api.publish.maven.MavenPublication
 
-void configureProject(Project project, String outputDir) {
+initscript {
+    repositories {
+        google()
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'org.yaml:snakeyaml:2.2+'
+    }
+}
+
+
+ext{
+    pubspecLock = new org.yaml.snakeyaml.Yaml().load(
+            new File(new File(System.getProperty('user.dir')).parentFile, "pubspec.lock").newInputStream()
```

Comment:
Defining the YAML instance here could reduce the number of file reads from disk caused by multiple calls to 'configureProject'.